common/src/chain/transaction/output.rs

```diff
@@ -4,8 +4,11 @@ use script::Script;
 
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Decode)]
 pub enum Destination {
+    #[codec(index = 0)]
     Address(crate::address::Address), // Address type to be added
```

Comment:
Yes. I think there's a misunderstanding. An address cannot be converted back to a public key, which is why there's two kinds of transactions. Same in bitcoin, P2PKH and P2PK. Hence, you cannot deduce the public key given an address, because it's generally a hash of the public key.

Reply:
The contention I have is that the `crate::address::Address` type contains a `bech32`-encoded string, which is not something I would expect to be included in blocks.

Reply:
That's fair. But regardless of whether we include the string itself or some other pre-bech32 form, there has to be distinction between addresses and public keys. I'll see what I can do.